commercial/app/controllers/HostedContentController.scala

```diff
@@ -49,6 +49,8 @@ class HostedContentController(
     } recover {
       case e: ContentApiError if e.httpStatus == 410 =>
         cached(commercialExpired(wasAHostedPage = true))
+      case e: ContentApiError if e.httpStatus == 404 =>
+        NoCache(NotFound)
```

Comment:
Looking over this again, I'm not sure `NoCache` is the right thing here. Is there a reason not to cache? I copied this from the catch all a few lines above, but this is probably closer to the 410 case just above, which does return cache headers. Any thoughts on this?

Reply:
I'm not familiar to this piece of code, but it seems odd that it used to work in the past, redirecting to a 404 without the explicit case for checking the httpStatus is a 404. Let's try that and see what happens?

Reply:
Thanks @ioanna0. Yeah it's weird, from the email thread I think maybe this always worked with regular articles, but perhaps it didn't for advertiser content like this (it sounds like a takedown of this kind of content is rare).